account-compression/programs/account-compression/src/instructions/update_address_merkle_tree.rs

```diff
@@ -0,0 +1,72 @@
+use account_compression_state::{address_merkle_tree_from_bytes_mut, address_queue_from_bytes_mut};
+use anchor_lang::prelude::*;
+use ark_ff::BigInteger256;
+use light_indexed_merkle_tree::array::{IndexingElement, RawIndexingElement};
+use light_utils::be_bytes_to_bigint;
+
+use crate::{
+    errors::AccountCompressionErrorCode,
+    state::address::{AddressMerkleTreeAccount, AddressQueueAccount},
+};
+
+#[derive(Accounts)]
+pub struct UpdateMerkleTree<'info> {
+    #[account(mut)]
+    pub authority: Signer<'info>,
+    #[account(mut)]
+    pub queue: AccountLoader<'info, AddressQueueAccount>,
+    #[account(mut)]
+    pub merkle_tree: AccountLoader<'info, AddressMerkleTreeAccount>,
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn process_update_address_merkle_tree<'info>(
+    ctx: Context<'_, '_, '_, 'info, UpdateMerkleTree<'info>>,
+    changelog_index: u16,
+    queue_index: u16,
+    address_index: u16,
+    address_next_index: u16,
+    address_next_value: [u8; 32],
+    low_address: RawIndexingElement<32>,
+    low_address_next_value: [u8; 32],
+    low_address_proof: [[u8; 32]; 40],
```

Comment:
I think this is not going to fit into one transaction.
2 * 4 + 32 + 66 + 32 + 40 * 32 = 1418

Reply:
We will need to add the zk proof (proving the Merkle proof) for the high element as well, which is an additional 128 bytes.

Reply:
I think the max we can do is height 23 if we want to do the update in one tx.
Otherwise we need to split it into two tx.
So let's split the update process into two tx (you can use the concept of the verifier state again).

Reply:
Additional idea, you could already mock the zkp as input data with [u8;128] just to get the size right.